packages/block-editor/src/components/ungroup-button/index.native.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * External dependencies
+ */
+import { noop } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import {
+	Toolbar,
+	ToolbarButton,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { withSelect, withDispatch } from '@wordpress/data';
+import { compose } from '@wordpress/compose';
+
+/**
+ * Internal dependencies
+ */
+import { Ungroup } from './icon';
+
+export function UngroupButton( {
+	onConvertFromGroup,
+	isUngroupable = false,
+} ) {
+	if ( ! isUngroupable ) {
+		return null;
+	}
+	return (
+		<Toolbar>
+			<ToolbarButton
+				title={ __( 'Ungroup' ) }
+				icon={ Ungroup }
+				onClick={ onConvertFromGroup }
+			/>
+		</Toolbar>
+	);
+}
+
+export default compose( [
+	withSelect( ( select ) => {
+		const {
+			getSelectedBlockClientId,
+			getBlock,
+		} = select( 'core/block-editor' );
+
+		const selectedId = getSelectedBlockClientId();
+		const selectedBlock = getBlock( selectedId );
+
+		const isUngroupable = selectedBlock && selectedBlock.innerBlocks && !! selectedBlock.innerBlocks.length;
+		const innerBlocks = isUngroupable ? selectedBlock.innerBlocks : [];
+
+		return {
+			isUngroupable,
+			clientId: selectedId,
+			innerBlocks,
+		};
+	} ),
+	withDispatch( ( dispatch, { clientId, innerBlocks, onToggle = noop } ) => {
```

Comment:
One more thing, I'm curious what this onToggle is used for?

Reply:
This button implementation comes from the web version and I believe that this is gateway to pass some action after ungroup.

I'm not sure how it is use on web and at this moment I didn't noticed any reason that we may want to pass callback on mobile version. I left it there for consistence with web but perhaps if your going to say that if we do not use it we should remove it I would agree...